sql-plugin/src/main/scala/org/apache/spark/sql/rapids/arithmetic.scala

```diff
@@ -144,6 +144,8 @@ object GpuDivModLike {
       case DType.INT64 => Scalar.fromLong(0L)
       case DType.FLOAT32 => Scalar.fromFloat(0f)
       case DType.FLOAT64 => Scalar.fromDouble(0)
+      case dt if dt.isDecimalType && dt.isBackedByInt => Scalar.fromDecimal(0, 0)
```

Comment:
I'm not sure this is going to work in all cases.  I think this might be another place were we have some tech debt to pay off and need to pass in a DataType instead of a DType.